packages/datadog_flutter_plugin/test/tracing/tracing_headers_test.dart

```diff
@@ -371,16 +603,129 @@ void main() {
     'traceparent tracing headers are generated correctly { TraceContextInjection.all, unsampled }',
     () {
       when(() => mockRum.shouldSampleTrace(any(), any())).thenReturn(false);
-      final context = generateTracingContext(mockRum);
+      final context = generateTracingContext(mockSdk, mockRum);
 
-      final headers = getTracingHeaders(
+      final headers = <String, String>{};
+      injectTracingHeaders(
         context,
         TracingHeaderType.tracecontext,
+        headers,
         contextInjection: TraceContextInjection.sampled,
       );
 
       expect(headers['traceparent'], isNull);
       expect(headers['tracestate'], isNull);
     },
   );
+
+  group('baggage header merging', () {
+    test('no baggage header creates new header', () {
+      // Given
+      String? oldBaggage;
+      final datadogContext = DatadogContext(
+        accountId: randomString(),
+        userId: randomString(),
+      );
+      final sessionId = randomString();
+      when(() => mockPlatform.cachedContext).thenReturn(datadogContext);
+      when(() => mockRum.cachedSessionId).thenReturn(sessionId);
+
+      when(() => mockRum.shouldSampleTrace(any(), any())).thenReturn(true);
+      final context = generateTracingContext(mockSdk, mockRum);
+
+      // When
+      String newBaggage = mergeW3CBaggageHeader(context, oldBaggage);
+
+      // Then
+      final baggageValues = newBaggage.split(',');
+      expect(baggageValues, contains('session.id=$sessionId'));
+      expect(baggageValues, contains('user.id=${datadogContext.userId}'));
+      expect(baggageValues, contains('account.id=${datadogContext.accountId}'));
+    });
+
+    test('existing baggage header adds new values', () {
+      // Given
+      String oldBaggage = 'test_value_1=1,test_value_2=string';
```

Comment:
FYI @marco-saia-datadog recently also added few tests https://github.com/DataDog/dd-sdk-reactnative/pull/1016/files#diff-395eac9b3950188084fb673149f381453a4f7fee342e91ed57f36ae94f0c3718, so maybe it is worth checking them as well.